pkg/sif/descriptor_input.go

```diff
@@ -253,6 +253,30 @@ func OptSBOMMetadata(f SBOMFormat) DescriptorInputOpt {
 	}
 }
 
+var errDigestTooLarge = errors.New("digest value too large")
+
+// OptOCIBlobMetadata sets metadata for a OCI blob data object.
+//
+// If this option is applied to a data object with an incompatible type, an error is returned.
+func OptOCIBlobMetadata(digest string) DescriptorInputOpt {
+	return func(t DataType, opts *descriptorOpts) error {
+		if got, want := t, DataOCIBlob; got != want {
+			return &unexpectedDataTypeError{got, []DataType{want}}
```

Comment:
I think we probably want to support this for `DataOCIRootIndex` also?

```suggestion
		if t != DataOCIRootIndex && t != DataOCIBlob {
			return &unexpectedDataTypeError{t, []DataType{DataOCIRootIndex, DataOCIBlob}}
```